infra/src/main/java/com/redhat/service/smartevents/infra/models/dto/ManagedResourceStatus.java

```diff
@@ -23,6 +26,16 @@ public String serialize() {
         return status;
     }
 
+    @SuppressWarnings("unused")
+    // Required for JAX-RS deserialisation. See @javax.ws.rs.QueryParam.
+    public static ManagedResourceStatus fromString(String status) {
+        return Arrays
+                .stream(ManagedResourceStatus.values())
```

Comment:
I remember this problem with the enums, but I think we should avoid having to duplicate this snippet all the time see also `ProcessorType`

Reply:
This can be also said of the duplicated code to serialise the enumerations too.

See [`ProcessorType`](https://github.com/5733d9e2be6485d52ffa08870cabdee0/sandbox/blob/main/infra/src/main/java/com/redhat/service/smartevents/infra/models/processors/ProcessorType.java#L11-L15) and [`ManagedResourceStatus`](https://github.com/5733d9e2be6485d52ffa08870cabdee0/sandbox/blob/48ea872cac9f67f1791b0f1f2af82d72748eeca9/infra/src/main/java/com/redhat/service/smartevents/infra/models/dto/ManagedResourceStatus.java#L20-L24).

The only way I can see to remove both of these duplications is to write a `javax.ws.rs.ext.ParamConverterProvider` however I'd need one for each enumeration and therefore I'm simply moving the duplication from one place to another. I am open to suggestions.